src/main/java/persistence/sql/ddl/CreateTableQueryBuilder.java

```diff
@@ -1,41 +1,43 @@
 package persistence.sql.ddl;
 
-import jakarta.persistence.*;
+import domain.Person;
+import persistence.sql.TableColumn;
+import persistence.sql.TableId;
 
-import java.util.Arrays;
+import java.util.List;
 import java.util.stream.Collectors;
 
 
 public class CreateTableQueryBuilder extends DDLQueryBuilder {
 
     private static final String CREATE_TABLE = "CREATE TABLE ";
 
-    @Override
-    public String executeQuery(Class<?> entityClass) {
-            return createTable(entityClass);
+    public CreateTableQueryBuilder(Class<?> entityClass) {
+        super(entityClass);
     }
 
-    public String createTable(Class<?> entityClass) {
-        if (!entityClass.isAnnotationPresent(Entity.class)) {
-            throw new IllegalArgumentException("This Class is not an Entity ");
-        }
-
-        String tableName = getTableName(entityClass);
-        String columns = Arrays.stream(entityClass.getDeclaredFields())
-                .filter(field -> !field.isAnnotationPresent(Transient.class)).map(field -> {
-                    StringBuilder sb = new StringBuilder();
-                    String columnName = getColumnName(field);
-                    Class<?> type = field.getType();
-
-                    sb.append(columnName).append(" ");
-                    sb.append(H2DBDataType.castType(type));
-
-                    appendColumnAttributes(field, sb);
-
-                    return sb.toString();
-                }).collect(Collectors.joining(", "));
+    @Override
+    public String executeQuery() {
+        return createTable();
+    }
 
-        return CREATE_TABLE + tableName + " (" + columns + ");";
+    private String createTable() {
+        List<TableColumn> tableColumn = tableMeta.getTableColumn();
+        TableId tableId = tableMeta.getTableId();
+        String idColumn = tableId.getName() + " " + H2DBDataType.castType(tableId.getType()) + " PRIMARY KEY" + (tableId.isAutoIncrement() ? " AUTO_INCREMENT, " : ", ");
+        String columns = tableColumn.stream().map(column -> {
+            StringBuilder sb = new StringBuilder();
+            sb.append(column.getColumnName()).append(" ");
+            sb.append(H2DBDataType.castType(column.getType()));
+            sb.append(column.isNotNullable() ? " NOT NULL" : "");
+            return sb.toString();
+        }).collect(Collectors.joining(", "));
+
+        return CREATE_TABLE + tableMeta.getTableName() + " ("+ idColumn + columns + ");";
```

Comment:
```suggestion
        String tableName = tableMeta.getTableName();
        String idColumn = createIdColumn(tableMeta.getTableId());
        String columns = createColumns(tableMeta.getTableColumn());

        return String.format("%s%s (%s%s);", CREATE_TABLE, tableName, idColumn, columns);
    }
```

좀 더 잘 읽히지 않나요?
함수가 복잡해진다면, 로직별로 메소드로 분리시켜보면 좋아요. 
중요한점은 로직이 의미하는것을 메소드 이름으로 잘 정의하는것이에요.

Reply:
따로 메서드로 분리하도록 하겠습니다!